packages/kbn-json-ast/src/snip.ts

```diff
@@ -25,7 +25,8 @@ export function snip(source: string, snips: Snip[]) {
     .reduce((acc: Snip[], s) => {
       const prev = acc.at(-1);
       if (!prev || prev[1] < s[0]) {
-        return [...acc, s];
+        acc.push(s);
+        return acc;
```

Comment:
There's a similar spread a few lines below. In the light of total elimination, should that be replaced by something like:
```typescript
      const merged: Snip = [Math.min(prev[0], s[0]), Math.max(prev[1], s[1])];
      acc[acc.length-1] = merged;
      return acc;
```